stories/utils/generate-widget-stories.js

```diff
@@ -26,59 +26,22 @@ import { Component } from 'react';
  * Internal dependencies
  */
 import Widgets from 'googlesitekit-widgets';
-import { STORE_NAME as CORE_SITE } from '../../assets/js/googlesitekit/datastore/site/constants';
-import { STORE_NAME as CORE_MODULES } from '../../assets/js/googlesitekit/modules/datastore/constants';
-import { WithTestRegistry } from '../../tests/js/utils';
-const { components: { Widget } } = Widgets;
-
-/**
- * Generates a function to set up registry for widget stories.
- *
- * @since 1.16.0
- *
- * @param {(string|Array)} moduleSlug Module slug or slugs to activate.
- * @param {string|null}    url        Current entity URL.
- * @param {Function}       [cb]       Callback for additional setup. Default is no-op.
- * @return {Function} A function to set up registry for widget stories.
- */
-function getSetupRegistry( moduleSlug, url, cb = () => {} ) {
-	return ( { dispatch } ) => {
-		cb( { dispatch } );
-
-		dispatch( CORE_SITE ).receiveSiteInfo( {
-			referenceSiteURL: null,
-			currentEntityURL: url,
-		} );
+import {
+	createTestRegistry,
+	WithTestRegistry,
+	provideModules,
+	provideSiteInfo,
+} from '../../tests/js/utils';
 
-		let modules = [];
-		if ( Array.isArray( moduleSlug ) ) {
-			modules = moduleSlug.map( ( module ) => {
-				return {
-					slug: module,
-					active: true,
-					connected: true,
-				};
-			} );
-		} else {
-			modules = [
-				{
-					slug: moduleSlug,
-					active: true,
-					connected: true,
-				},
-			];
-		}
-		dispatch( CORE_MODULES ).receiveGetModules( modules );
-	};
-}
+const { components: { Widget } } = Widgets;
 
 /**
  * Generates stories for a report based widget using provided data.
  *
  * @since 1.16.0
  *
  * @param {Object}    args                              Widget arguments.
- * @param {string}    args.moduleSlug                   Module slug.
+ * @param {Array}     args.moduleSlug                   List of modules to activate.
```

Comment:
I'm okay making this require an array, but in that case it should also be renamed accordingly, to `moduleSlugs`. `moduleSlug` for an array variable doesn't really make sense.